okta/cache/cache.go

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2018 - Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cache
+
+import (
+	"net/http"
+)
+
+type Cache interface {
+	Get(key string) *http.Response
+	Set(key string, value *http.Response)
+	Delete(key string)
+	Clear()
+	Has(key string) bool
+}
+
+func CreateCacheKey(req *http.Request) string {
```

Comment:
Perhaps `CreateCacheKeyFromRequest` to be more specific? I'm also not sure what the value is in leaving off the protocol?  If I were inspecting/debugging my cache, I think it would be nice to have the full URL.

Reply:
I will consider this for the future